modules/apps/segments/segments-test/src/testIntegration/java/com/liferay/segments/odata/retriever/test/UserODataRetrieverCustomFieldsTest.java

```diff
@@ -241,6 +241,76 @@ public void testGetUsersFilterByCustomFieldWithEqualsAndDateTextType()
 		Assert.assertEquals(user1, users.get(0));
 	}
 
+	@Test
+	public void testGetUsersFilterByCustomFieldWithEqualsAndIntegerKeywordType()
+		throws Exception {
+
+		ExpandoColumn expandoColumn = _addExpandoColumn(
+			_expandoTable, RandomTestUtil.randomString(),
+			ExpandoColumnConstants.INTEGER,
+			ExpandoColumnConstants.INDEX_TYPE_KEYWORD);
+
+		int columnValue = 3;
+
+		User user1 = _addUser(expandoColumn.getName(), columnValue);
+
+		_users.add(user1);
+
+		User user2 = UserTestUtil.addUser();
+
+		_users.add(user2);
+
+		String filterString = String.format(
+			"(customField/%s eq %s)", _encodeName(expandoColumn), columnValue);
+
+		int count = _getODataRetriever().getResultsCount(
+			TestPropsValues.getCompanyId(), filterString,
+			LocaleUtil.getDefault());
+
+		Assert.assertEquals(1, count);
+
+		List<User> users = _getODataRetriever().getResults(
+			TestPropsValues.getCompanyId(), filterString,
+			LocaleUtil.getDefault(), 0, 1);
+
+		Assert.assertEquals(user1, users.get(0));
+	}
+
+	@Test
+	public void testGetUsersFilterByCustomFieldWithEqualsAndIntegerTextType()
```

Comment:
I'd add a test for `ExpandoColumnConstants.SHORT` and another for `ExpandoColumnConstants.LONG`, does it make sense for you?